sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -225,17 +229,25 @@ impl InputSelection {
             // PANIC: safe to unwrap as non basic/account/foundry/nft outputs are already filtered out.
             let unlock_conditions = input.output.unlock_conditions().unwrap();
 
-            if unlock_conditions.is_time_locked(self.slot_index) {
+            if unlock_conditions.is_timelocked(self.slot_index, self.protocol_parameters.min_committable_age()) {
                 return false;
             }
 
             let required_address = input
                 .output
                 // Account transition is irrelevant here as we keep accounts anyway.
-                .required_and_unlocked_address(self.slot_index, input.output_id())
-                // PANIC: safe to unwrap as non basic/account/foundry/nft outputs are already filtered out.
-                .unwrap()
-                .0;
+                .required_address(
+                    self.slot_index,
+                    self.protocol_parameters.min_committable_age(),
+                    self.protocol_parameters.max_committable_age(),
```

Comment:
maybe we want to extract a `CommittableRange { min_age: _, max_age: _ }` type, returned by a method on `ProtocolParameters` to this a bit more convenient?

Reply:
Good call